### PR TITLE
e2e test for action metrics

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

End to end tests for this PR https://github.com/MetaMask/metamask-extension/pull/18346

### CODE REVIEW

File: test/e2e/mv3/service-worker-restart.spec.js
Comment (digiwand) [inline test/e2e/mv3/service-worker-restart.spec.js:180]: I think I missed something. I thought `service_worker_action_queue_methods` was an array. how is it a number here and an array in other places?

**Review by digiwand — COMMENTED** [review]
nice tests! left a couple of comments. looks like we might want to handle https://github.com/MetaMask/metamask-extension/issues/18530 before merging this one?

File: test/e2e/mv3/service-worker-restart.spec.js
Comment (darkwing) [inline test/e2e/mv3/service-worker-restart.spec.js:165]: Total nit, not a blocker, but you could do `const [firstResult]  = request.body.json.batch;` so you can avoid the long `request.body.json.batch[0]` everywhere.

File: app/scripts/metamask-controller.js
Comment (darkwing) [inline app/scripts/metamask-controller.js:1273]: Good catch!

**Review by darkwing — DISMISSED** [review]

File: test/e2e/mock-e2e.js
Comment (danjm) [inline test/e2e/mock-e2e.js:70]: @pedronfigueiredo remind me: why was this change needed in this PR?

**Review by danjm — DISMISSED** [review]
LGTM!

I asked a question. Might be worth adding a code comment on that sentry mock, but this is not a blocker

File: development/build/index.js
Comment (Gudahtt) [inline development/build/index.js:135]: Hmm. This will have a significant impact upon the production bundle. Making this change in a PR that appears to be test-focused is potentially misleading.

When creating a changelog, we might reasonably look at this PR and conclude no functional changes were made.

**Review by DDDDDanica — DISMISSED** [review]

**Review by brad-decker — CHANGES_REQUESTED** [review]
Requesting changes to block merge until @Gudahtt 's comment is resolved

**Review by brad-decker — CHANGES_REQUESTED** [review]
Just notating one last blocker

**Review by DDDDDanica — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]
LGTM

**Pull request merged**
